### DISCUSSION TITLE
Can I use multiple hooks on one input type?

### DISCUSSION THREAD

**Discussion by Diluka** [TriPSs/nestjs-query#139] [Q&A] [answered]
I want to add a common hook and a specific hook on an input type. I don't want to create many hooks with a lot of duplicate codes.

**Comment by TriPSs** [comment, marked as answer]
Yes, you can do the following:
```ts
@BeforeCreateOne(HookOne, HookTwo)
```

**Reply by Diluka** [reply]
it works. I didn't find relevant examples in the docs.

**Reply by TriPSs** [reply]
Correct this is quite new feature that we forgot to add to the docs.

**Reply by Diluka** [reply]
By the way, if I want to inject a Service, what does the configuration of the Service look like in the module.

**Reply by TriPSs** [reply]
You would need to add it to the `services`: 
```ts
NestjsQueryGraphQLModule.forFeature({
      imports: [],
      services: [
        // Here
      ],
      resolvers: []
})
```

**Reply by Diluka** [reply]
Then the Hook class also needs to be added here?

**Reply by TriPSs** [reply]
No the hook class does not have to be added there.

**Reply by Diluka** [reply]
If `MyService` already added to `MyModule`'s `providers`, and I can't remove it from `providers`. Can I add it both `MyModule. providers` and `services`?

**Reply by TriPSs** [reply]
Think that should work yes.
